hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/SingleTransactionBlockItems.java

```diff
@@ -0,0 +1,103 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+
+/**
+ * A logical transaction wrapper for the block items produced for/by processing a single
+ * transaction.
+ *
+ * @param txn    the submitted user transaction
+ * @param result the result of processing the user transaction
+ * @param output the output (if any) of processing the user transaction
+ */
+public record SingleTransactionBlockItems(
+        @NonNull Transaction txn, @NonNull TransactionResult result, @Nullable TransactionOutput output) {
+
+    /**
+     * The input block items should be exactly the block items produced for/by processing a
+     * single transaction, with the following expected order:
+     * <ol>
+     *     <li>Index 0: BlockItem.Transaction</li>
+     *     <li>Index 1: BlockItem.TransactionResult</li>
+     *     <li>Index 2: BlockItem.TransactionOutput (if applicable)</li>
+     * </ol>
+     *
+     * @param items The block items representing a single transaction
+     * @return A logical transaction wrapper for the block items
+     */
+    public static SingleTransactionBlockItems asSingleTransaction(@NonNull final List<BlockItem> items) {
+        final var builder = new Builder();
+        final var txnItem = items.get(0);
+        if (!txnItem.hasTransaction()) {
+            throw new IllegalArgumentException("Expected a transaction item!");
+        }
+        // The nullable warnings here aren't warranted since we check for `hasTransaction()` above.
+        // Similarly for the other fields
+        //noinspection DataFlowIssue
+        builder.txn(txnItem.transaction());
+
+        final var resultItem = items.get(1);
+        if (!resultItem.hasTransactionResult()) {
+            throw new IllegalArgumentException("Expected a transaction result item!");
+        }
+        //noinspection DataFlowIssue
+        builder.result(resultItem.transactionResult());
+
+        if (items.size() > 2) {
+            final var maybeOutput = items.get(2);
+            if (!maybeOutput.hasTransactionOutput()) {
+                throw new IllegalArgumentException("Expected a transaction output item!");
+            }
+            builder.output(maybeOutput.transactionOutput());
+        }
+
+        return builder.build();
+    }
+
+    public static class Builder {
+        private Transaction txn;
+        private TransactionResult result;
+        private TransactionOutput output;
+
+        public Builder txn(@NonNull final Transaction txn) {
+            this.txn = txn;
+            return this;
+        }
+
+        public Builder result(@NonNull final TransactionResult result) {
+            this.result = result;
+            return this;
+        }
+
+        public Builder output(@Nullable TransactionOutput output) {
+            this.output = output;
+            return this;
+        }
+
+        public SingleTransactionBlockItems build() {
```

Comment:
I've sometimes wondered why the `build` method of builders doesn't do the verification of all fields.  E.g., non-null.  It just isn't generally done, that I've seen, but why not?

Reply:
Good point!

Reply:
Done